### PR TITLE
Add performPrep.py and tests 

### PR DESCRIPTION
**Pull request by rbechto2**
This pull request combines work from Aamna, Liang, and Raphael to create performPrep.py and test_performPrep.py

### CODE REVIEW

**Comment by adam2392** [comment]
Okay it looks good. Try to link the codes into a cohesive pipeline and you can just run one integration test that tests the entire thing (removes redundancy). 

- link entire prep with ransac with robust referencing, etc. 
- include a jupyter notebook tutorial of what is going on here
- consider encapsulating functions that fall under same category under a class. See staticmethods, classmethods in python. 

^ for all above see graspy

File: pyautomagic/preprocessing/prep/noisychannels.py [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+        channel_deviation[i] = 0.7413 * iqr(EEGData[i, :])
+    channel_deviationSD = 0.7413 * iqr(channel_deviation)
+    channel_deviationMedian = np.nanmedian(channel_deviation)
+    robust_channel_deviation[evaluation_channels] = np.divide(
+        np.subtract(channel_deviation, channel_deviationMedian), channel_deviationSD
+    )
+    for i in range(0, new_dimension[0]):
+        deviation_channel_mask[i] = abs(robust_channel_deviation[i]) > 5 or np.isnan(
+            robust_channel_deviation[i]
+        )
```
Comment (Nick3151) [inline pyautomagic/preprocessing/prep/noisychannels.py]: Dimensions seems not matching, robust_channel_deviation has original_dimension but i only goes through new_dimension

**Review by adam2392 — COMMENTED** [review]
Is "tests/test_data/S001R04.edf" necessary? Do you want to include this as a test file into the overall repo? If not, remove it from the PR.

Overall, I'm assuming this is to integrate the "PREP" pipeline functionality into pyautomagic right? If so, perhaps it would be helpful to get this merged in asap. @Nick3151 and @AamnaLawrence is this functional and is the code how you would like it? Can you insert your pyprep easily once it is done to wherever Raph has it set up?

I'll leave it up you two to approve this PR, and have it merged in. Then we can see if additional changes need to be made, and you two can submit issues related to that.

**Review by adam2392 — CHANGES_REQUESTED** [review]
Is "tests/test_data/S001R04.edf" necessary? Do you want to include this as a test file into the overall repo? If not, remove it from the PR. If so, ignore this.

**Comment by adam2392** [comment]
What is this PR vs the #13 one?

**Pull request closed without merging**